src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -155,6 +155,60 @@ private[spark] case class CosmosDBConnection(config: Config) extends CosmosDBLog
     returnValue.get
   }
 
+  private def tryHandlePartitionSplitError
+  (
+    outerError: Throwable,
+    error: DocumentClientException,
+    originalFeedOptions: FeedOptions,
+    counterSnapshot: Int
+  ) : (Int, Array[String]) = {
+    if (counterSnapshot > 1000 ||
+      error.getStatusCode() != 410) {
```

Comment:
only one nit, maybe it just won't ever happen, but could the `error` in the case of the Illegal exception be `null` or empty`?

Reply:
No lines 253-255 prevent that - In the new connector I would use an assert to clarify - but in the old connector we haven't wired up asserts. And this method is private - so I think it is ok